query/undefine.rs

```diff
@@ -4,22 +4,1149 @@
+            type_: Label<'static>,
+            value_type: ValueType,
+            existing_value_type: ValueType,
+            declaration: CapabilityType
+        ),
+        RelatesSpecialiseNotDefined(
+            29,
+            "Undefining 'as {specialised_role_name}' for '{type_} relates {specialising_role_name}' failed since there is no defined '{type_} relates {specialising_role_name} as {specialised_role_name}'.\nSource:\n{declaration}",
+            type_: Label<'static>,
+            specialising_role_name: String,
+            specialised_role_name: String,
+            declaration: Specialise
+        ),
+        RelatesSpecialiseDefinedButDifferent(
+            30,
+            "Undefining 'as {specialised_role_name}' for '{type_} relates {specialising_role_name}' failed since there is no defined '{type_} relates {specialising_role_name} as {specialised_role_name}', while there is a defined specialisation '{type_} relates {specialising_role_name} as {existing_specialised_role_name}'.\nSource:\n{declaration}",
+            type_: Label<'static>,
+            specialising_role_name: String,
+            specialised_role_name: String,
+            existing_specialised_role_name: String,
+            declaration: Specialise
+        ),
+        TypeAnnotationNotDefined(
+            31,
+            "Undefining annotation '{annotation}' for type '{type_}' failed since there is no defined '{type_} {annotation}'.\nSource:\n{declaration}",
+            type_: Label<'static>,
+            annotation: AnnotationCategory,
+            declaration: AnnotationType
+        ),
+        CapabilityAnnotationNotDefined(
+            32,
+            "Undefining annotation '{annotation}' for a capability failed since there is no defined annotation of this category.\nSource:\n{declaration}",
+            annotation: AnnotationCategory,
+            declaration: AnnotationCapability
+        ),
+        IllegalAnnotation(
+            33,
+            "Illegal annotation",
+            ( source: AnnotationError )
+        ),
+        CapabilityKindMismatch(
+            34,
+            "Undefining failed because the left type '{left}' is of kind '{left_kind}' isn't the same kind as the right type '{right}' which has kind '{right_kind}'. Maybe you wanted to undefine something else?\nSource:\n{declaration}",
+            left: Label<'static>,
+            right: Label<'static>,
+            declaration: CapabilityType,
+            left_kind: Kind,
+            right_kind: Kind
+        ),
     }
 );
+
+impl fmt::Display for UndefineError {
+    fn fmt(&self, _: &mut fmt::Formatter<'_>) -> fmt::Result {
+        todo!()
+    }
+}
+
+impl Error for UndefineError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        todo!()
+    }
+}
```

Comment:
`TypeDBError` doesn't require these, and has its own impl of `Display` based on provided description. These should be removed.

Reply:
Huh, I was sure I had an error when I tried removing it (and I saw that it's already implemented in the macro). Now it works, so it's good. Thanks